Update hdf5/package.py for two HDF5 CMake variable names to be changed in HDF5 2.0.0

lrknox opened the pull request:
Update ALLOW_UNSUPPORTED to HDF5_ALLOW_UNSUPPORTED and HDF5_ENABLE_Z_LIB_SUPPORT to HDF5_ENABLE_ZLIB_SUPPORT for HDF5 2.0.0 and above to match the change in HDF5.

These changes are in the develop-2.0 version.

lrknox committed:
Update ALLOW_UNSUPPORTED to HDF5_UNSUPPORTED and
HDF5_ENABLE_Z_LIB_SUPPORT to HDF5_ENABLE_ZLIB_SUPPORT for HDF5 2.0.0 and
above to match the change in HDF5.

lrknox committed:
Merge branch 'develop' into update-hdf5-vars

tldahlgren commented:
Added missing maintainers.

tldahlgren commented:
Restarted just the failed vtk@9.3.1 job on windows stack that timed out. There seems to be a container issue already though.

brtnfld commented on repos/spack_repo/builtin/packages/hdf5/package.py:
```diff
@@ -552,6 +547,22 @@ def cmake_args(self):
+
         api = spec.variants["api"].value
         if api != "default":
             args.append(self.define("DEFAULT_API_VERSION", api))
```
DEFAULT_API_VERSION was changed to HDF5_DEFAULT_API_VERSION. So this needs to be updated (not tested):

```
api = spec.variants["api"].value
if api != "default":
    if self.spec.satisfies("@2.0.0:"):
        args.append(self.define("HDF5_DEFAULT_API_VERSION", api))
    else:
        args.append(self.define("DEFAULT_API_VERSION", api))
```

lrknox replied:
Updated and tested manually

brtnfld commented on repos/spack_repo/builtin/packages/hdf5/package.py:
```diff
@@ -552,6 +547,22 @@ def cmake_args(self):
+        else:
+            args.append(self.define("ALLOW_UNSUPPORTED", True))
+
+        # The name of option Z_LIB_SUPPORT was also changed to ZLIB_SUPPORT.
```
for clarity: HDF5_ENABLE_Z_LIB_SUPPORT was changed to HDF5_ENABLE_ZLIB_SUPPORT.

lrknox replied:
updated accordingly

brtnfld reviewed (changes requested):
DEFAULT_API_VERSION needs fixing

brtnfld commented on repos/spack_repo/builtin/packages/hdf5/package.py:
```diff
@@ -552,6 +547,22 @@ def cmake_args(self):
             self.define_from_variant("HDF5_BUILD_TOOLS", "tools"),
         ]
 
+        # Always enable this option. This does not actually enable any
```
...specify certain combinations of other arguments that would normally be rejected as unsupported configurations. Note: This option was renamed from ALLOW_UNSUPPORTED to HDF5_ALLOW_UNSUPPORTED in HDF5 2.0.0 as part of the standardization of CMake option naming.

lrknox replied:
Updated accordingly.

lrknox committed:
Added handling of DEFAULT_API_VERSION to HDF5_DEFAULT_API_VERSION in
version 2.0.
Replaced never released v116 api with v200.
Improved some comments.

lrknox committed:
Merge branch 'update-hdf5-vars' of https://github.com/lrknox/spack-packages into update-hdf5-vars

lrknox committed:
Remove trailing whitespace.

lrknox committed:
Merge branch 'develop' into update-hdf5-vars

tldahlgren commented:
@brtnfld Are you okay with this PR being merged with the changes? (You're still flagged as requesting changes.)

lrknox commented:
@tldahlgren, 3 test configurations failed because "The git server, Gitaly, is not available at this time", and 3 others because vtk@9.3.1 had this error:  "ERROR: Job failed: execution took longer than 6h0m0s seconds".  The first error might not happen if those test configurations run again, but I'm not seeing a way for me to rerun them.  Also, I don't think I can do anything about the vtk test length, and I don't think it is connected to the changes to this PR.  Are you able to do anything about these errors?

lrknox committed:
Merge branch 'develop' into update-hdf5-vars
